distributed/tests/test_worker_state_machine.py

```diff
@@ -1002,6 +1003,138 @@ def test_gather_priority(ws):
     ]
 
 
+def test_running_task_in_all_running_tasks(ws_with_running_task):
+    ws = ws_with_running_task
+    task = ws.tasks["x"]
+    assert task
+    assert task in ws.all_running_tasks
+
+
+def test_cancelled_running_task_in_all_running_tasks(ws_with_running_task):
+    ws = ws_with_running_task
+
+    ws.handle_stimulus(FreeKeysEvent(keys=["x"], stimulus_id="cancel"))
+    task = ws.tasks["x"]
+    assert task
+    assert task.state == "cancelled"
+    assert task in ws.all_running_tasks
+
+
+def test_resumed_running_task_in_all_running_tasks(ws_with_running_task):
+    ws = ws_with_running_task
+
+    ws.handle_stimulus(
+        FreeKeysEvent(keys=["x"], stimulus_id="cancel"),
+        ComputeTaskEvent.dummy(
+            key="y",
+            who_has={"x": ["127.0.0.1:1235"]},
+            nbytes={"x": 8},
+            stimulus_id="compute-y",
+        ),
+    )
+    task = ws.tasks["x"]
+    assert task
+    assert task.state == "resumed"
+    assert task in ws.all_running_tasks
+
+
+@pytest.mark.xfail(reason="distributed#6565")
+def test_successful_running_task_not_in_all_running_tasks(ws_with_running_task):
+    ws = ws_with_running_task
```

Comment:
```suggestion
    ws = ws_with_running_task
    ts = ws.tasks["x"]
    assert ts in ws.all_running_tasks

```

Reply:
Testing the same thing in multiple places is not a good idea, unless it helps improving clarity.
In this case, it makes it absolutely clear to the reader that the task was there to begin with and it left, instead of forcing them to figure out if another test is checking for it.